### PR TITLE
use try_fopen from psr7 package

### PR DESCRIPTION
**Pull request by gmponos**

### CODE REVIEW

File: src/Handler/StreamHandler.php [hunk 1]
```diff
@@ -152,8 +152,7 @@ private function createSink(StreamInterface $stream, array $options): StreamInte
             return $stream;
         }
 
-        $sink = $options['sink']
-            ?? \fopen('php://temp', 'r+');
+        $sink = $options['sink'] ?? Psr7\try_fopen('php://temp', 'r+');
```
Comment (Nyholm) [inline src/Handler/StreamHandler.php:155]: In what scenario can't you open `php://temp`?

File: src/Handler/StreamHandler.php [hunk 2]
```diff
@@ -327,7 +326,7 @@ static function () use ($context, $params) {
 
         return $this->createResource(
             function () use ($uri, &$http_response_header, $contextResource, $context, $options, $request) {
-                $resource = \fopen((string) $uri, 'r', false, $contextResource);
+                $resource = @\fopen((string) $uri, 'r', false, $contextResource);
```
Comment (Nyholm) [inline src/Handler/StreamHandler.php:329]: Why?

File: src/Utils.php [hunk 3]
```diff
@@ -73,12 +73,7 @@ public static function debugResource($value = null)
             return \STDOUT;
         }
 
-        $resource = \fopen('php://output', 'w');
-        if (false === $resource) {
-            throw new \RuntimeException('Can not open php output for writing to debug the resource.');
-        }
-
-        return $resource;
+        return \GuzzleHttp\Psr7\try_fopen('php://output', 'w');
```
Comment (Nyholm) [inline src/Utils.php:76]: Hm..

File: src/Handler/CurlFactory.php [hunk 4]
```diff
@@ -412,7 +412,7 @@ private function applyHandlerOptions(EasyHandle $easy, array &$conf): void
             };
         } else {
             // Use a default temp stream if no sink was set.
-            $conf[\CURLOPT_FILE] = \fopen('php://temp', 'w+');
+            $conf[\CURLOPT_FILE] = Psr7\try_fopen('php://temp', 'w+');
```
Comment (GrahamCampbell) [inline src/Handler/CurlFactory.php:415]: Need to use the new static methods.

File: src/Handler/StreamHandler.php (hunk 1 above)
Comment (GrahamCampbell) [reply src/Handler/StreamHandler.php:155]: Need to use the new static methods.

File: src/Utils.php (hunk 3 above)
Comment (GrahamCampbell) [reply src/Utils.php:76]: Need to use the new static methods.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
Replaced by https://github.com/guzzle/guzzle/pull/2876.

**Pull request closed without merging**